### PR TITLE
feature/217 propare bulk_get_or_create to taipy-rest

### PR DESCRIPTION
**Pull request by toan-quach**

### CODE REVIEW

**Review by khaitruong922 — DISMISSED** [review]

File: src/taipy/rest/api/resources/job.py [hunk 1]
```diff
@@ -173,7 +173,7 @@ def post(self):
     def __create_job_from_schema(self, task_name: str) -> Optional[Job]:
         task_manager = TaskManager()
         try:
-            task = task_manager._get_or_create(self.fetch_config(task_name))
+            task = task_manager._bulk_get_or_create([self.fetch_config(task_name)])[0]
```
Comment (jrobinAV) [inline src/taipy/rest/api/resources/job.py:176]: I believe bulk_get_or_create returns a dictionary (config/entity). I believe `[0]` does not allow you to retrieve the task.

File: src/taipy/rest/api/resources/task.py [hunk 2]
```diff
@@ -160,7 +160,7 @@ def post(self):
 
         try:
             config = self.fetch_config(config_id)
-            task = manager._get_or_create(config)
+            task = manager._bulk_get_or_create([config])[0]
```
Comment (jrobinAV) [inline src/taipy/rest/api/resources/task.py:163]: same here.

**Review by jrobinAV — CHANGES_REQUESTED** [review]

File: src/taipy/rest/api/resources/job.py (hunk 1 above)
Comment (khaitruong922) [reply src/taipy/rest/api/resources/job.py:176]: @jrobinAV  I believe it is returning a list of tasks.

Comment (khaitruong922) [reply src/taipy/rest/api/resources/job.py:176]: The one you are showing is of DataManager. This is the signature of TaskManager. I am not sure if we should have inconsistent signature between managers.

```python
    def _bulk_get_or_create(
        cls,
        task_configs: List[TaskConfig],
        scenario_id: Optional[ScenarioId] = None,
        pipeline_id: Optional[PipelineId] = None,
    ) -> List[Task]:
```

Comment (jrobinAV) [reply src/taipy/rest/api/resources/job.py:176]: Oh my bad I was not looking at the right manager method !

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**